lib/cisco_node_utils/bgp.rb

```diff
@@ -351,6 +351,7 @@ def confederation_peers
     def confederation_peers=(should)
       # confederation peers are additive. So create a delta hash of entries
       # and only apply the changes
+      should = should.flatten
```

Comment:
Is there any check that you should do here to ensure the flatten is appropriate? Something like:

```ruby
should = should.flatten if should[0].is_a?(Array)
```